cli/tests/pcluster/schemas/test_cluster_schema.py

```diff
@@ -575,3 +580,76 @@ def test_scheduler_plugin_scheduler_definition_schema(plugin_interface_version,
         assert_that(scheduler_plugin_definition.events.head_init.execute_command.command).is_equal_to(
             events["HeadInit"]["ExecuteCommand"]["Command"]
         )
+
+
+@pytest.mark.parametrize(
+    "scheduler_definition, grant_sudo_privileges, s3_error, https_error, failure_message",
+    [
+        ("s3://bucket/scheduler_definition.yaml", True, None, None, None),
+        (
+            "s3://bucket/scheduler_definition_fake.yaml",
+            True,
+            AWSClientError(function_name="get_object", message="The specified key does not exist."),
+            None,
+            "Error while downloading scheduler definition from 's3://bucket/scheduler_definition_fake.yaml': The specified key does not exist.",
+        ),
+        ("https://bucket.s3.us-east-2.amazonaws.com/scheduler_definition.yaml", False, None, None, None),
+        (
+            "https://bucket.s3.us-east-2.amazonaws.com/scheduler_definition_fake.yaml",
+            None,
+            None,
+            HTTPError(
+                url="https://test-slurm.s3.us-east-2.amazonaws.com/scheduler_definition_fake.yaml",
+                code=403,
+                msg="Forbidden",
+                hdrs="dummy",
+                fp="dummy",
+            ),
+            "Error while downloading scheduler definition from 'https://bucket.s3.us-east-2.amazonaws.com/scheduler_definition_fake.yaml': HTTP Error 403: Forbidden",
+        ),
+        (
+            {"PluginInterfaceVersion": "1.0", "Events": {"HeadInit": {"ExecuteCommand": {"Command": "env"}}}},
+            True,
+            None,
+            None,
+            None,
+        ),
+    ],
+)
+def test_scheduler_plugin_settings_schema(
```

Comment:
can you also test what happens when the text you try to decode is not a valid YAML? is the error clear enough or we need to more granularly handle such case?